### PR TITLE
Alerting: No longer index state history log streams by instance labels

### PR DESCRIPTION
**Pull request by alexweav**
**What is this feature?**

This is a continuation of https://github.com/grafana/grafana/pull/65403 where we stop merging instance labels into the log stream labels. Going forward, log streams correspond 1:1 to rules.

Queries for labels will rely on the functionality added in https://github.com/grafana/grafana/pull/65403, they now filter the log line itself rather than labels.

**Why do we need this feature?**

1. Big reduction in cardinality.
2. Easier to reason about limits. Number of streams = number of rules, which already has limits.
3. Stream labels are well-known and predictable.

**Which issue(s) does this PR fix?**:

n/a

**Special notes for your reviewer:**

# Consider reviewing this PR commit-by-commit

Please check that:
- [x] It works as expected from a user's perspective.
- [x] If this is a pre-GA feature, it is behind a feature toggle.
- [x] The docs are updated, and if this is a [notable improvement](https://grafana.com/docs/writers-toolkit/writing-guide/contribute-release-notes/#how-to-determine-if-content-belongs-in-a-whats-new-document), it's added to our [What's New](https://grafana.com/docs/writers-toolkit/writing-guide/contribute-release-notes/) doc.
- [x] There are no known compatibility issues with older supported versions of Grafana, or plugins.
- [x] It passes the [Hosted Grafana feature readiness review](https://docs.google.com/document/d/1QL9Ly8KnXzpb6ISbg49pTODRO5mhA5tkkfIZVX6pqQU/edit) for observability, scalability, performance, and security.

### CODE REVIEW

File: pkg/services/ngalert/state/historian/loki.go [hunk 1]
```diff
@@ -343,3 +313,60 @@ func jsonifyRow(line string) (json.RawMessage, error) {
...
+	return "{" + query + "}"
+}
+
+func isValidOperator(op string) bool {
+	switch op {
+	case "=", "!=", "=~", "!~":
+		return true
+	}
+	return false
+}
```
Comment (alexweav) [inline pkg/services/ngalert/state/historian/loki.go:352]: copied verbatim from `loki_http.go`

File: pkg/services/ngalert/state/historian/loki_test.go [hunk 2]
```diff
@@ -180,6 +140,70 @@ func TestRemoteLokiBackend(t *testing.T) {
...
+	t.Run("new selector", func(t *testing.T) {
+		selector, err := NewSelector("label", "=", "value")
+		require.NoError(t, err)
+		require.Equal(t, "label", selector.Label)
+		require.Equal(t, Eq, selector.Op)
+		require.Equal(t, "value", selector.Value)
+
+		selector, err = NewSelector("label", "invalid", "value")
+		require.Error(t, err)
+	})
```
Comment (alexweav) [inline pkg/services/ngalert/state/historian/loki_test.go:165]: copied verbatim from separate test functions

File: pkg/services/ngalert/state/historian/loki_test.go [hunk 3]
```diff
@@ -65,55 +65,16 @@ func TestRemoteLokiBackend(t *testing.T) {
...
 				"folderUID":          rule.NamespaceUID,
 				"group":              rule.Group,
 				"orgID":              fmt.Sprint(rule.OrgID),
 				"ruleUID":            rule.UID,
-				"a":                  "b",
 			}
-			require.Equal(t, exp, res[0].Stream)
-		})
-
-		t.Run("groups streams based on combined labels", func(t *testing.T) {
```
Comment (alexweav) [inline pkg/services/ngalert/state/historian/loki_test.go:82]: not needed any more. the type system now enforces this automatically

**Review by gotjosh — APPROVED** [review]
LGTM

**Review by JohnnyQQQQ — APPROVED** [review]
LGTM

**Pull request merged**